src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1224,10 +1224,15 @@ private boolean loadAlias(final Resource resource, Map<String, Map<String, Colle
             containingResource = resource;
         }
 
+        if (containingResource == null) {
```

Comment:
I would move this check a bit up into the block after line 1222:

```
if (JCR_CONTENT.equals(resource.getName())) {
  containingResource = resource.getParent();
  if (containingResource == null) {
    log.warn(...)
    return false;
  }
} else ...
```

because only there ``containingResource`` can be assigned that null value.

Reply:
Sounds right.